### PR TITLE
fix(ngOptions) skip authentic empty options when looking for options

### LINKED ISSUE #12190: ngOptions in Angular 1.4 broke <select> placeholder values using ng-if
In Angular, you can specify a placeholder value for a `<select>` by doing something like so:

```
      <select ng-options="o.name for o in options" ng-model="model">
        <option value="">Please choose</option>
      </select>
```

This still works, however even when you choose an option in the list, the 'Please choose' placeholder still shows. To get around this, you could use an `ng-if` on the placeholder `<option>` to hide the option if a model value was selected, like so:

```
      <select ng-options="o.name for o in options" ng-model="model">
        <option ng-if="!model" value="">Please choose</option>
      </select>
```

However this no longer works in Angular 1.4, the placeholder no longer shows and the entire list is cleared if you choose one of the other options.

I have created a plunkr to demonstrate this, see here: http://plnkr.co/edit/NDXZLkGavwBU3Qg5YAtT

### PR DESCRIPTION
**Pull request by petebacondarwin**
Related #12952
Closes #12190
Closes #13029

### CODE REVIEW

File: src/ng/directive/ngOptions.js [hunk 1]
```diff
@@ -618,10 +618,19 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         var emptyOption_ = emptyOption && emptyOption[0];
         var unknownOption_ = unknownOption && unknownOption[0];
 
+        // If the compiled empty option was replaced by a comment because
+        // it had an "element" transclusion directive on it (such as ngIf)
+        // then compare against the pre-compiled empty element instead
+        if (emptyOption && emptyOption_.nodeType === NODE_TYPE_COMMENT) {
```
Comment (Narretz) [inline src/ng/directive/ngOptions.js]: We can actually remove this whole block then, can't we? (the tests pass without it)

Comment (Narretz) [inline src/ng/directive/ngOptions.js]: Because of `providedEmptyOption = emptyOption`, providedEmptyOption is also the compiled result. But the added checks below do the trick.

Comment (petebacondarwin) [inline src/ng/directive/ngOptions.js]: I guess so. As long as there is not some crazy `replace` directive on the option :-)

Comment (Narretz) [inline src/ng/directive/ngOptions.js]: Even that works in its simplest form, I just tested it. :) This case is probably very rare anyway, because having an option that replaces itself with an option takes this whole issue to strange new places!

ShallI remove the block and commit?

Comment (gkalpak) [inline src/ng/directive/ngOptions.js]: If this block is removed, don't forget to also undo the change on [line 411][1].

[1]: https://github.com/angular/angular.js/pull/13033/files#diff-0a9e86a58b2663074f51454f7a18a0efR411

**Comment by Narretz** [comment]
I'll clean this up like proposed and merge it, if that's okay for you guys.

**Comment by petebacondarwin** [comment]
@Narretz sounds good
